### PR TITLE
Add protos for runner pool state

### PR DESCRIPTION
**Pull request by bduffany**
Add protos to make the runner pool state serializable so that we can save it to a file and then reconstruct the runner pool state from the file after an executor restart.

Supporting only firecracker for now, but in theory we could also support resuming bare/podman/docker persistent workers after a restart.

**Related issues**: N/A

### CODE REVIEW

**Review by maggie-lou — APPROVED** [review]

File: proto/runner.proto [hunk 1]
```diff
@@ -87,3 +88,60 @@ message RunResponse {
...
+  RunnerKey runner_key = 1;
+
+  // State specific to the isolation type in use.
+  ContainerState container_state = 2;
+
+  // Short debug ID used to identify the runner.
+  string debug_id = 3;
+
+  // Counter that is incremented each time the runner is assigned a new task.
+  int64 task_number = 4;
```
Comment (maggie-lou) [inline proto/runner.proto:129]: nit: Rename to `assigned_task_count`? `number` sounds a bit like an ID

Comment (bduffany) [reply proto/runner.proto:129]: much better - done

File: proto/runner.proto [hunk 2]
```diff
@@ -87,3 +88,60 @@ message RunResponse {
...
+  // Runner key.
+  RunnerKey runner_key = 1;
+
+  // State specific to the isolation type in use.
+  ContainerState container_state = 2;
+
+  // Short debug ID used to identify the runner.
+  string debug_id = 3;
+
+  // Counter that is incremented each time the runner is assigned a new task.
```
Comment (vadimberezniker) [inline proto/runner.proto:128]: What's the need for this field?

**Review by vadimberezniker — APPROVED** [review]

File: proto/runner.proto (hunk 2 above)
Comment (bduffany) [reply proto/runner.proto:128]: It provides some signal about the health of the runner pool when looking at logs. If this value is very low on average, it means we're hitting the pool limits too frequently or failing to recycle too frequently.

**Pull request merged**